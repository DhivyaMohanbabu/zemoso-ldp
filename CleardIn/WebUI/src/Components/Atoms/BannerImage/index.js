import React from 'react';
import { makeStyles } from '@material-ui/core/styles';
import CardMedia from '@material-ui/core/CardMedia';
import PropTypes from 'prop-types';

const useStyles = makeStyles((theme) => ({
  image: {
    margin: 'auto',
    [theme.breakpoints.down('xs')]: {
      height: '150px',
      width: '176px',
      objectFit: 'cover',
    },
    [theme.breakpoints.up('sm')]: {
      height: '170px',
      width: '225px',
      objectFit: 'cover',
    },
    [theme.breakpoints.up('md')]: {
      height: '264px',
      width: '310px',
      objectFit: 'cover',
    },
    [theme.breakpoints.up('lg')]: {
      height: '350px',
      width: '406px',
      objectFit: 'cover',
    },
  },
}));

export default function BannerImage({ title, imageSrc }) {
  const classes = useStyles();
  const imageSrcSet = `${imageSrc[1]} 2x, ${imageSrc[2]} 3x`;
  return (
    <CardMedia
      className={classes.image}
      component="img"
      title={title}
      src={imageSrc[0]}
      srcSet={imageSrcSet}
    />
  );
}

BannerImage.propTypes = {
  title: PropTypes.string.isRequired,
  imageSrc: PropTypes.arrayOf(PropTypes.string).isRequired,
};
