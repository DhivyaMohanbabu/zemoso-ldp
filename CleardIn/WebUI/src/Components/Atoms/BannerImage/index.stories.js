import React from 'react';
import svgImg from '../../../../assets/images/access.svg';
import BannerImage from '.';

export default {
  title: 'Atom/BannerImage',
};
const imageSrcForSVG = [svgImg];
const Template = (args) => <BannerImage {...args} />;

export const Default = Template.bind({});
Default.args = {
  title: 'Sample SVG Image',
  imageSrc: imageSrcForSVG,
};
