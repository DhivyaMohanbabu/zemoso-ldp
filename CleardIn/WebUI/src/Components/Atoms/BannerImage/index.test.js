import React from 'react';
import '@testing-library/jest-dom';
import { render } from '@testing-library/react';
import pngImg from '../../../../assets/images/access_logo.png';
import BannerImage from '.';

const allProps = {
  title: 'Sample Test',
  imageSrc: [pngImg],
};

it('renders BannerImage snapshot correctly', () => {
  const { container } = render(<BannerImage {...allProps} />);
  expect(container).toMatchSnapshot();
});
