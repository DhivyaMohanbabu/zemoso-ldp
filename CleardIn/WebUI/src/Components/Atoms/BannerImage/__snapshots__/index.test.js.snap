// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders BannerImage snapshot correctly 1`] = `
<div>
  <img
    class="MuiCardMedia-root makeStyles-image-1 MuiCardMedia-media MuiCardMedia-img"
    src="test-file-stub"
    srcset="undefined 2x, undefined 3x"
    title="Sample Test"
  />
</div>
`;
